app/src/screens/Splash.tsx

```diff
@@ -185,179 +109,43 @@ const Splash = () => {
-        // add post authentication screens
-        const postAuthScreens = []
-        if (store.onboarding.didAgreeToTerms !== TermsVersion) {
-          postAuthScreens.push(Screens.Terms)
-        }
-        if (!store.onboarding.didConsiderPushNotifications && enablePushNotifications) {
-          postAuthScreens.push(Screens.UsePushNotifications)
-        }
-        dispatch({ type: DispatchAction.SET_POST_AUTH_SCREENS, payload: [postAuthScreens] })
-
-        if (!store.loginAttempt.lockoutDate) {
-          navigation.dispatch(
-            CommonActions.reset({
-              index: 0,
-              routes: [{ name: Screens.EnterPIN }],
-            })
-          )
-        } else {
-          // return to lockout screen if lockout date is set
-          navigation.dispatch(
-            CommonActions.reset({
-              index: 0,
-              routes: [{ name: Screens.AttemptLockout }],
-            })
-          )
-        }
-
-        return
-      }
-
-      // If onboarding was interrupted we need to pickup from where we left off.
-      navigation.dispatch(
-        CommonActions.reset({
-          index: 0,
-          routes: [
-            {
-              name: resumeOnboardingAt(
-                store.onboarding.didSeePreface,
-                store.onboarding.didCompleteTutorial,
-                store.onboarding.didAgreeToTerms,
-                store.onboarding.didCreatePIN,
-                store.onboarding.didNameWallet,
-                store.onboarding.didConsiderBiometry,
-                store.onboarding.didConsiderPushNotifications,
-                TermsVersion,
-                store.preferences.enableWalletNaming,
-                showPreface,
-                !!enablePushNotifications
-              ),
-            },
-          ],
-        })
-      )
-    } catch (e: unknown) {
-      setInitErrorType(InitErrorTypes.Onboarding)
-      setInitError(new BifoldError(t('Error.Title2030'), t('Error.Message2030'), (e as Error)?.message, 2030))
+    if (initializing.current) {
+      return
     }
-  }, [
-    mounted,
-    setStep,
-    store.authentication.didAuthenticate,
-    store.stateLoaded,
-    store.onboarding.onboardingVersion,
-    store.onboarding.didCompleteOnboarding,
-    store.onboarding.didSeePreface,
-    store.onboarding.didCompleteTutorial,
-    store.onboarding.didAgreeToTerms,
-    store.onboarding.didCreatePIN,
-    store.onboarding.didConsiderPushNotifications,
-    store.onboarding.didNameWallet,
-    store.onboarding.didConsiderBiometry,
-    store.preferences.enableWalletNaming,
-    enablePushNotifications,
-    showPreface,
-    dispatch,
-    store.loginAttempt.lockoutDate,
-    navigation,
-    initOnboardingCount,
-    t,
-  ])
 
-  useEffect(() => {
+    setStep(1)
+    setStep(2)
+
     const initAgentAsyncEffect = async (): Promise<void> => {
       try {
-        if (
-          !mounted ||
-          initializing.current ||
-          !store.authentication.didAuthenticate ||
```

Comment:
I think we still need just the `!store.authentication.didAuthenticate` check here as per this PR:
https://github.com/openwallet-foundation/bifold-wallet/pull/1477

Reply:
Fixed

Reply:
Just curious, is there a reason you moved the early return after the `checkForUpdates()` call? Might cause it to be called many times

Reply:
I'm on the fence with this one. You can't get to the Splash screen unless you complete either the PIN Create or PIN Enter which both complete authentication. So if you're seeing Splash you must have completed authentication. So, on 2nd though - this change may not have been needed. Thoughts?

I moved it after because it should be cached at this point and `didAuthenticate` is only required for the agent. I think it would be better to move the initialized check up if we don't want to repeat this fn call. Thoughts?